File: java-backend/src/main/java/com/baklawati/backend/models/Shipping.java
// File: src/main/java/com/baklawati/backend/models/Shipping.java

package com.baklawati.backend.models;

import jakarta.persistence.*;

@Entity
@Table(name = "shipping")
public class Shipping {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String address;
    private String city;
    private String state;
    private String postalCode;
    private String country;

    @OneToOne
    @JoinColumn(name = "order_id")
    private Order order;

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getAddress() { return address; }
    public void setAddress(String address) { this.address = address; }

    public String getCity() { return city; }
    public void setCity(String city) { this.city = city; }

    public String getState() { return state; }
    public void setState(String state) { this.state = state; }

    public String getPostalCode() { return postalCode; }
    public void setPostalCode(String postalCode) { this.postalCode = postalCode; }

    public String getCountry() { return country; }
    public void setCountry(String country) { this.country = country; }

    public Order getOrder() { return order; }
    public void setOrder(Order order) { this.order = order; }
}

File: flutter-app/app/lib/models/shipping.dart
class Shipping {
  final int id;
  final int orderId;
  final String address;
  final String city;
  final String postalCode;
  final String country;
  final String status;

  Shipping({
    required this.id,
    required this.orderId,
    required this.address,
    required this.city,
    required this.postalCode,
    required this.country,
    required this.status,
  });

  factory Shipping.fromJson(Map<String, dynamic> json) {
    return Shipping(
      id: json['id'],
      orderId: json['order_id'],
      address: json['address'],
      city: json['city'],
      postalCode: json['postal_code'],
      country: json['country'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'order_id': orderId,
      'address': address,
      'city': city,
      'postal_code': postalCode,
      'country': country,
      'status': status,
    };
  }
}
